amqp/src/main/scala/akka/stream/alpakka/amqp/AmqpConnectionProvider.scala

```diff
@@ -21,6 +21,16 @@ sealed trait AmqpConnectionProvider {
   def release(connection: Connection): Unit = if (connection.isOpen) connection.close()
 }
 
+/**
```

Comment:
This should be moved to the test project with the `AmqpProxyConnection`since it's just a test utility.

Reply:
Actually, I don't even think that you need this provider at all.
You could use the `AmqpConnectionFactoryConnectionProvider` and pass to it a connection factory that returns your spied connection.

Something like:
```
    val factory = new ConnectionFactory {
      override def newConnection(): Connection = new AmqpProxyConnection(conn)
    }
```